sample-apps/payment-customization-sample-app/web/frontend/pages/index.jsx

```diff
@@ -0,0 +1,138 @@
+import {
+  Page,
+  Card,
+  TextStyle,
+  EmptyState,
+  IndexTable,
+  useIndexResourceState,
+  Link,
+} from "@shopify/polaris";
+import { PlusMinor } from "@shopify/polaris-icons";
+import { useIsMutating } from 'react-query'
+
+import {
+  usePaymentCustomizations,
+  useDeletePaymentCustomization,
+} from "../hooks";
+
+export default function HomePage() {
+  const { data = [], isFetching, refetch } = usePaymentCustomizations();
+
+  const { mutateAsync: deleteCustomization } = useDeletePaymentCustomization();
+
+  const isMutating = useIsMutating(["deleteCustomization"]);
+
+  const {
+    selectedResources,
+    allResourcesSelected,
+    handleSelectionChange,
+    clearSelection,
+  } = useIndexResourceState(data);
+
+  const handleDeleteAction = async () => {
+    if (isMutating) return
+
+    try {
+      const mutations = selectedResources.map(id => deleteCustomization({ params: { id }}))
+      await Promise.all(mutations)
+      await refetch();
+      clearSelection();
+    }
+    catch (error) {
+      console.error(error)
+    }
+  }
+
+  const primaryAction = {
+    content: "Create customization",
+    icon: PlusMinor,
+    url: "/new",
+  }
+
+  const resourceName = {
+    singular: "customization",
+    plural: "customizations",
+  }
+
+  const tableHeadings = [
+    {title: 'ID'},
+    {title: 'Payment Method'},
+    {title: 'Cart Subtotal'},
+  ]
+
+  const tableActions = [
+    {
+      content: 'Delete customizations',
+      onAction: handleDeleteAction
+    },
+  ]
+
+  const selectedCount = allResourcesSelected ? 'All' : selectedResources.length
+  const isLoading = isFetching || isMutating
+
+  return (
+    <Page
+      title="Customizations"
+      primaryAction={primaryAction}
+    >
+      <Card>
+        <IndexTable
+          resourceName={resourceName}
+          itemCount={data.length}
+          headings={tableHeadings}
+          promotedBulkActions={tableActions}
+          loading={isLoading}
+          emptyState={<EmptyTable action={primaryAction} />}
```

Comment:
The empty state card is still rendered when the Spinner is active. 
Should this be `emptyState={ !isLoading && <EmptyTable action={primaryAction} />}`